src/managers/PlayerManager.ts

```diff
@@ -46,6 +47,36 @@ class PlayerManager {
         );
     }
 
+    public getRoles(): string[] {
+        const seed: string = GeneralManager.getSeed();
+        const allRoles: string[] = this.generateRoles();
+        const shuffledRoles: string[] = seededShuffle(allRoles, seed);
+        return shuffledRoles;
+    }
+
+    public generateRoles(): string[] {
+        let roles: string[] = [];
+        const session: Nullable<SessionInformation> =
+            GeneralManager.getSession();
+        if (session === null) {
+            throw new Error("Couldn't get Session");
+        }
+        const playerCount: int = session.playerCount;
+        if (playerCount <= 4) {
+            roles.push("Saboteur");
+        } else if (playerCount <= 6) {
+            roles.push("Saboteur", "Saboteur");
+        } else if (playerCount <= 9) {
+            roles.push("Saboteur", "Saboteur", "Saboteur");
+        } else {
+            roles.push("Saboteur", "Saboteur", "Saboteur", "Saboteur");
+        }
```

Comment:
Couldn't this be done in a smart formula way... with input playerCount and output saboteur count.

Reply:
Not without if-else statements. That's why thought I'd do it like this (very simple and slightly ugly). But rather directly pushing onto the array, I could just have number of saboteurs and then at the end do a loop filling the array up...

Reply:
Ive added a statement to lobbyscreen that only allows player to start game if 4 or more people are in session. So id suggest 
if playerCount == 4, return saboteurCount = 1
else:
saboteurCount = 1+( (playerCount-1) /3), assuming floor division

Reply:
I would suggest
if (playerCount = 4) return 1
else:
saboteurCount = 1+( (playerCount-1) /3), assuming floor division

Reply:
This will probably be redesigned in future commit. For now, if it works it works.